instance/models/utils.py

```diff
@@ -482,6 +484,49 @@ def purge(self):
         """
         return self._client.kv.delete(self.prefix, recurse=True)
 
+    def leave(self, force=True):
+        """
+        This method will request the agent to leave the cluster, if the
+        operation wasn't completed successfully then it'll force nodes to
+        leave the cluster if force option is set to True.
+
+        :param force: Will fall back to the force mode if the operation
+                      wasn't completed successfuly.
+        :return: True if the operation completed successfully, False otherwise.
```

Comment:
It's not currently doing that, because if the server `is_left` (successful graceful leave), we skip the next `if` check and just return `None` by default. We can assign `is_left` to `self.force_leave` in that conditional and then ultimately return `is_left`, or whatever gets the job done.